ofrak_core/ofrak/core/openwrt.py

```diff
@@ -212,29 +212,32 @@ async def unpack(self, resource: Resource, config=None):
         # Create lzma loader child
         await resource.create_child(
             tags=(OpenWrtTrxLzmaLoader,),
-            data=data[trx_header.trx_loader_offset : trx_header.trx_kernel_offset],
+            data=data[trx_header.trx_loader_offset: trx_header.trx_kernel_offset],
         )
         # Create kernel child
         await resource.create_child(
             tags=(OpenWrtTrxKernel,),
-            data=data[trx_header.trx_kernel_offset : trx_header.trx_rootfs_offset],
+            data=data[
+                 trx_header.trx_kernel_offset: trx_header.trx_rootfs_offset] if trx_header.trx_rootfs_offset > 0 else data[
+                                                                                                                      trx_header.trx_kernel_offset:],
         )
         if trx_version == OpenWrtTrxVersion.VERSION1:
             # Create rootfs child
             await resource.create_child(
                 tags=(OpenWrtTrxRootfs,),
-                data=data[trx_header.trx_rootfs_offset :],
+                data=data[trx_header.trx_rootfs_offset:] if trx_header.trx_rootfs_offset > 0 else [],
             )
         else:  # TRX Version 2
             # Create rootfs child
             await resource.create_child(
                 tags=(OpenWrtTrxRootfs,),
-                data=data[trx_header.trx_rootfs_offset : trx_header.trx_binheader_offset],
+                data=data[
+                     trx_header.trx_rootfs_offset: trx_header.trx_binheader_offset] if trx_header.trx_rootfs_offset > 0 else [],
```

Comment:
The data should be an empty bytestring `b""`, not an empty list `[]`, when rootfs is null